### PR TITLE
ReCAPTCHA verification error handling

### PR DESCRIPTION
**Pull request by protocol86**
Adds error handling for errors returned from the response URL during a reCAPTCHA app verification flow.

### CODE REVIEW

File: Firebase/Auth/Source/FIRAuthErrorUtils.h [hunk 1]
```diff
@@ -460,12 +460,27 @@ NS_ASSUME_NONNULL_BEGIN
...
+    @return The NSError instance associated with the given FIRAuthError.
+ */
++ (NSError *)appVerificationUserInteractionFailureWithReason:(nullable NSString *)reason;
+
+/** @fn reCAPTCHAResponseErrorWithCode:message:
+    @brief Constructs an @c NSError with the code and message provided.
+    @param message Error message from the backend, if any.
+    @return The NSError instance associated with the given FIRAuthError.
+ */
++ (NSError *)URLResponseErrorWithCode:(NSString *)code message:(nullable NSString *)message;
```
Comment (XiangtianDai) [inline Firebase/Auth/Source/FIRAuthErrorUtils.h]: returns `nullable NSError *`.

File: Firebase/Auth/Source/FIRAuthErrorUtils.h [hunk 2]
```diff
@@ -460,12 +460,27 @@ NS_ASSUME_NONNULL_BEGIN
...
+        FIRAuthErrorCodeAppVerificationUserInteractionFailure code.
+    @param reason Reason for error, returned via URL response.
+    @return The NSError instance associated with the given FIRAuthError.
+ */
++ (NSError *)appVerificationUserInteractionFailureWithReason:(nullable NSString *)reason;
+
+/** @fn reCAPTCHAResponseErrorWithCode:message:
+    @brief Constructs an @c NSError with the code and message provided.
+    @param message Error message from the backend, if any.
+    @return The NSError instance associated with the given FIRAuthError.
```
Comment (XiangtianDai) [inline Firebase/Auth/Source/FIRAuthErrorUtils.h]: @return The NSError instance associated with the given error message, if one is found.

File: Firebase/Auth/Source/FIRAuthErrorUtils.h (hunk 1 above)
Comment (protocol86) [reply Firebase/Auth/Source/FIRAuthErrorUtils.h]: done.

File: Firebase/Auth/Source/FIRAuthErrorUtils.h (hunk 2 above)
Comment (protocol86) [reply Firebase/Auth/Source/FIRAuthErrorUtils.h]: done.

File: Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m [hunk 3]
```diff
@@ -158,15 +157,16 @@ - (void)verifyPhoneNumber:(NSString *)phoneNumber
...
             return;
           }
-          NSDictionary<NSString *, NSString *> *URLQueryItems =
-              [NSDictionary gtm_dictionaryWithHttpArgumentsString:callbackURL.query];;
-          URLQueryItems =
-              [NSDictionary gtm_dictionaryWithHttpArgumentsString:URLQueryItems[@"deep_link_id"]];
-          NSString *reCAPTCHA = URLQueryItems[@"recaptchaToken"];
+          NSError *reCAPTCHAError;
+          NSString *reCAPTCHAToken = [self reCAPTCHATokenForURL:callbackURL error:&reCAPTCHAError];
+          if (reCAPTCHAError) {
```
[the remark this answers is not in the document]
Comment (protocol86) [reply Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: Done, thanks.

**Pull request merged**